tests/test_object.h

```diff
@@ -87,6 +187,119 @@ TEST_CASE("[Object] Metadata") {
 			meta_list2.size() == 0,
 			"The metadata list should contain 0 items after removing all metadata items.");
 }
+
+TEST_CASE("[Object] Construction") {
+	Object object;
+
+	CHECK_MESSAGE(
+			!object.is_reference(),
+			"Object is not a Reference.");
+
+	Object *p_db = ObjectDB::get_instance(object.get_instance_id());
+	CHECK_MESSAGE(
+			p_db == &object,
+			"The database pointer returned by the object id should reference same object.");
+}
+
+TEST_CASE("[Object] Script instance property setter") {
+	Object object;
+	_MockScriptInstance script_instance;
+	object.set_script_instance(&script_instance);
```

Comment:
This seems to resolve the crash above:

```suggestion
	_MockScriptInstance *script_instance = memnew(_MockScriptInstance);
	object.set_script_instance(script_instance);
```

Script instance will be freed automatically by the `object`'s destructor, which actually caused the crash.

Generally, you'll also likely want to instantiate `Object` with `memnew` as well for consistency with the rest of the codebase (just like `Node`s), but don't forget to `memdelete` those as well.

The same applies to other test cases.